### PR TITLE
feat: Add rule for AIP-152 Jobs resource reference suffix

### PR DESCRIPTION
**Pull request by acamadeo**
Part of #698.

### CODE REVIEW

File: docs/rules/0152/request-resource-suffix.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+rule:
+  aip: 152
+  name: [core, '0152', request-resource-suffix]
+  summary: |
+
```
Comment (apasel422) [inline docs/rules/0152/request-resource-suffix.md:6]: Is this blank line needed?

File: go.sum [hunk 2]
```diff
@@ -47,6 +47,7 @@ github.com/kr/text v0.1.0 h1:45sCR5RtlFHMR4UwH9sdQ5TC8v0qDQCHnXt+kaKSTVE=
 github.com/kr/text v0.1.0/go.mod h1:4Jbv+DJW3UT/LiOwJeYQe1efqtUx/iVham/4vfdArNI=
 github.com/lithammer/dedent v1.1.0 h1:VNzHMVCBNG1j0fh3OrsFRkVUwStdDArbgBWoPAffktY=
 github.com/lithammer/dedent v1.1.0/go.mod h1:jrXYCQtgg0nJiN+StA2KgR7w6CiQNv9Fd/Z9BP0jIOc=
+github.com/mattn/go-runewidth v0.0.9 h1:Lm995f3rfxdpd6TSmuVCHVb/QhupuXlYr8sCI/QdE+0=
```
Comment (apasel422) [inline go.sum:50]: I don't think the changes in this `go.sum` belong in this PR.

File: rules/aip0152/aip0152.go [hunk 3]
```diff
@@ -35,6 +35,7 @@ func AddRules(r lint.RuleRegistry) error {
 		requestNameField,
 		requestNameReference,
 		responseMessageName,
+		requestResourceSuffix,
```
Comment (apasel422) [inline rules/aip0152/aip0152.go:38]: Let's keep these sorted.

**Review by apasel422 — CHANGES_REQUESTED** [review]

File: go.sum (hunk 2 above)
Comment (acamadeo) [reply go.sum]: Removed these changes. One note, without them when I try to run the tests I get the following error:
```
$ go test ./...
go: github.com/jhump/protoreflect@v1.8.1 requires
        google.golang.org/grpc@v1.27.0 requires
        golang.org/x/oauth2@v0.0.0-20180821212333-d2e6202438be: missing go.sum entry; to add it:
        go mod download golang.org/x/oauth2
```
Should this be addressed separately?

File: docs/rules/0152/request-resource-suffix.md (hunk 1 above)
Comment (acamadeo) [reply docs/rules/0152/request-resource-suffix.md]: No, removed.

File: rules/aip0152/aip0152.go (hunk 3 above)
Comment (acamadeo) [reply rules/aip0152/aip0152.go:37]: Done.

**Review by lukesneeringer — APPROVED** [review]
@apasel422 Can you clear your "changes requested" flag (unless you want more changes)?

File: go.sum (hunk 2 above)
Comment (apasel422) [reply go.sum]: Yes, they'll be handled in a separate PR.

**Review by apasel422 — APPROVED** [review]

**Comment by apasel422** [comment]
> @apasel422 Can you clear your "changes requested" flag (unless you want more changes)?

Done.

**Pull request merged**